samples/startup-script/index.js

```diff
@@ -16,130 +16,123 @@
-function createVm(name, callback) {
+async function createVm(name) {
   // Create a new VM, using default ubuntu image. The startup script
   // installs apache and a custom homepage.
-  const config = {
-    os: 'ubuntu',
-    http: true,
-    metadata: {
-      items: [
-        {
-          key: 'startup-script',
-          value: `#! /bin/bash
 
-        # Installs apache and a custom homepage
-        apt-get update
-        apt-get install -y apache2
-        cat <<EOF > /var/www/html/index.html
-        <!doctype html>
-        <h1>Hello World</h1>
-        <p>This page was created from a simple start-up script!</p>`,
-        },
-      ],
-    },
-  };
+  try {
+    const config = {
+      os: 'ubuntu',
+      http: true,
+      metadata: {
+        items: [
+          {
+            key: 'startup-script',
+            value: `#! /bin/bash
+    
+            # Installs apache and a custom homepage
+            apt-get update
+            apt-get install -y apache2
+            cat <<EOF > /var/www/html/index.html
+            <!doctype html>
+            <h1>Hello World</h1>
+            <p>This page was created from a simple start-up script!</p>`,
+          },
+        ],
+      },
+    };
+    const vmObj = zone.vm(name);
+    console.log('Creating VM ...');
+    const [vm, operation] = await vmObj.create(config);
+    await operation.promise();
+    const [metadata] = await vm.getMetadata();
 
-  const vm = zone.vm(name);
+    // External IP of the VM.
+    const ip = metadata.networkInterfaces[0].accessConfigs[0].natIP;
+    console.log(`Booting new VM with IP http://${ip}...`);
 
-  vm.create(config)
-    .then(data => {
-      const operation = data[1];
-      return operation.promise();
-    })
-    .then(() => {
-      return vm.getMetadata();
-    })
-    .then(data => {
-      const metadata = data[0];
-
-      // External IP of the VM.
-      const ip = metadata.networkInterfaces[0].accessConfigs[0].natIP;
-      console.log(`Booting new VM with IP http://${ip}...`);
-
-      // Ping the VM to determine when the HTTP server is ready.
-      let waiting = true;
-      const timer = setInterval(
-        ip => {
-          http
-            .get('http://' + ip, res => {
-              const statusCode = res.statusCode;
-              if (statusCode === 200 && waiting) {
-                waiting = false;
-                clearTimeout(timer);
-                // HTTP server is ready.
-                console.log('Ready!');
-                callback(null, ip);
-              }
-            })
-            .on('error', () => {
-              // HTTP server is not ready yet.
-              process.stdout.write('.');
-            });
-        },
-        2000,
-        ip
-      );
-    })
-    .catch(err => callback(err));
+    // Ping the VM to determine when the HTTP server is ready.
+    await pingVM(ip);
+    return ip;
+  } catch (err) {
+    console.error(`Something went wrong while creating ${name} :`, err);
```

Comment:
I don't think we need this try/catch.  Can we just let it throw?

Reply:
Done